### PR TITLE
Fix godep-restore

### PR DESCRIPTION
**Pull request by sttts**

### CODE REVIEW

File: hack/godep-save.sh [hunk 1]
```diff
@@ -9,13 +9,25 @@ pin-godep() {
   popd > /dev/null
 }
 
+# fail early without jq
+which jq >/dev/null || { echo "Please install jq." 1>&2; exit 1; }
```
Comment (stevekuznetsov) [inline hack/godep-save.sh]:
```
os::util::ensure::system_binary_exists "jq"
```

File: hack/godep-save.sh [hunk 2]
```diff
@@ -60,9 +72,11 @@ fork::with::fake::packages () {
   pushd "$TMPGOPATH/src/$PKG" >/dev/null
     local OLDREV=$(git rev-parse HEAD)
     for FAKEPKG in "$@"; do
+      test -n "$(ls $FAKEPKG/*.go 2>/dev/null)" && { echo "'fork::with::fake::packages $PKG $FAKEPKG' failed because $FAKEPKG already exists." 1>&2; exit 1; }
```
Comment (stevekuznetsov) [inline hack/godep-save.sh]: Prefer to use `if` instead of `test && ||`

File: hack/lib/cleanup.sh [hunk 3]
```diff
@@ -95,7 +95,7 @@ function os::cleanup::tmpdir() {
     fi
 
 		# delete any sub directory underneath BASETMPDIR
-    for directory in $( find "${BASETMPDIR}" -d 2 ); do
+    for directory in $( find "${BASETMPDIR}" -depth 2 ); do
```
Comment (stevekuznetsov) [inline hack/lib/cleanup.sh]: You're going to get a conflict on this I think, someone is currently fixing it.

**Comment by mfojtik** [comment]
@ncdc i think we should start kicking `godep` tool out from origin repo and switch to `golang/dep`... every release we adding more hacks/band-aid to survive. Maybe we can join forces on this (PM + your team) .

**Comment by ncdc** [comment]
Only if dep actually works. But I agree. The godep author has stated it is
basically eol.

**Review by stevekuznetsov — CHANGES_REQUESTED** [review]

File: hack/lib/cleanup.sh (hunk 3 above)
Comment (sttts) [reply hack/lib/cleanup.sh]: Will drop this change. There is still some error messages, might be Mac related.

Comment (stevekuznetsov) [reply hack/lib/cleanup.sh]: https://github.com/openshift/origin/pull/13691

**Pull request merged**